src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/SeedTemporaryAccommodationPremisesTest.kt

```diff
@@ -0,0 +1,270 @@
+    }
+
+    withCsv(
+      "invalid-local-authority-ta",
+      temporaryAccommodationPremisesSeedCsvRowsToCsv(
+        listOf(
+          TemporaryAccommodationPremisesSeedCsvRowFactory()
+            .withProbationRegion(probationRegion.name)
+            .withLocalAuthorityArea("Not Real Authority")
+            .produce()
+        )
+      )
+    )
+
+    seedService.seedData(SeedFileType.temporaryAccommodationPremises, "invalid-local-authority-ta")
+
+    assertThat(logEntries).anyMatch {
+      it.level == "error" &&
+        it.message == "Error on row 1:" &&
+        it.throwable != null &&
+        it.throwable.message == "Local Authority Area Not Real Authority does not exist"
+    }
+  }
+
+  @Test
+  fun `Creating a new Temporary Accommodation Premises persists correctly`() {
+    val probationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    val localAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
+
+    val csvRow = TemporaryAccommodationPremisesSeedCsvRowFactory()
+      .withProbationRegion(probationRegion.name)
+      .withLocalAuthorityArea(localAuthorityArea.name)
+      .withCharacteristics(listOf("Park nearby", "Not suitable for arson offenders"))
+      .produce()
+
+    withCsv(
+      "new-ta-premises",
+      temporaryAccommodationPremisesSeedCsvRowsToCsv(
+        listOf(
+          csvRow
+        )
+      )
+    )
+
+    seedService.seedData(SeedFileType.temporaryAccommodationPremises, "new-ta-premises")
+
+    val persistedPremises = temporaryAccommodationPremisesRepository.findByName(csvRow.name)
+    assertThat(persistedPremises).isNotNull
+    assertThat(persistedPremises!!.addressLine1).isEqualTo(csvRow.addressLine1)
+    assertThat(persistedPremises.addressLine2).isEqualTo(csvRow.addressLine2)
+    assertThat(persistedPremises.town).isEqualTo(csvRow.town)
+    assertThat(persistedPremises.postcode).isEqualTo(csvRow.postcode)
+    assertThat(persistedPremises.probationRegion.name).isEqualTo(csvRow.probationRegion)
+    assertThat(persistedPremises.localAuthorityArea!!.name).isEqualTo(csvRow.localAuthorityArea)
+    assertThat(persistedPremises.pdu).isEqualTo(csvRow.pdu)
+    assertThat(persistedPremises.characteristics.map { it.name }).isEqualTo(csvRow.characteristics)
+    assertThat(persistedPremises.notes).isEqualTo(csvRow.notes)
+  }
+
+  @Test
+  fun `Updating an existing Temporary Accommodation Premises persists correctly`() {
+    val originalProbationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    val updatedProbationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    val originalLocalAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
+
+    val updatedLocalAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
```

Comment:
It looks like we're using region and local authority as the fields we check to see if our updates were made on an existing record.

Do `probationRegionEntityFactory.produceAndPersist` and  `localAuthorityEntityFactory.produceAndPersist()` always return different values?

Might it be better to check for updates on a more straightforward field such as `notes` or `postcode`?

Reply:
This test adds a characteristic that isn't originally present on the premises ("Floor level access") and tests for its presence after the update.

I believe that `*Factory.produceAndPersist` does produce a different object each time as otherwise it wouldn't be able to insert it into the database as they'd have the same value for the primary key.